### PR TITLE
[release-1.0]ddl: update the column's offset when we do modify column

### PR DESCRIPTION
**Pull request by zimulala**
We should get the latest offset from the store when we do modify the column.

### CODE REVIEW

**Comment by jackysp** [comment]
LGTM

**Comment by zimulala** [comment]
/run-all-tests  tidb-test=release-1.0 tidb-private-test=release-1.0 tikv=release-1.0

File: ddl/column.go [hunk 1]
```diff
@@ -450,6 +450,8 @@ func (d *ddl) doModifyColumn(t *meta.Meta, job *model.Job, col *model.ColumnInfo
 		return ver, infoschema.ErrColumnNotExists.GenByArgs(oldName, tblInfo.Name)
 	}
 
+	col.Offset = oldCol.Offset
```
Comment (winkyao) [inline ddl/column.go]: rename col to newCol

File: ddl/ddl_api.go [hunk 2]
```diff
@@ -1259,9 +1259,8 @@ func (d *ddl) getModifiableColumnJob(ctx context.Context, ident ast.Ident, origi
 	}
 
 	newCol := table.ToColumn(&model.ColumnInfo{
-		ID:                 col.ID,
-		Offset:             col.Offset,
-		State:              col.State,
+		ID: col.ID,
```
Comment (winkyao) [inline ddl/ddl_api.go]: Please add some comments why we not set offset here.

File: ddl/column.go (hunk 1 above)
Comment (winkyao) [reply ddl/column.go]: Please add some comments why we set offset here.

**Comment by winkyao** [comment]
Plz fix ci

File: ddl/ddl_api.go (hunk 2 above)
Comment (zimulala) [reply ddl/ddl_api.go]: I think to add the comment to line454 in `column.go` is enough.

File: ddl/column.go [hunk 3]
```diff
@@ -450,6 +450,9 @@ func (d *ddl) doModifyColumn(t *meta.Meta, job *model.Job, col *model.ColumnInfo
 		return ver, infoschema.ErrColumnNotExists.GenByArgs(oldName, tblInfo.Name)
 	}
 
+	// We need the latest column's offset and state. This information can be obtained from.
```
Comment (winkyao) [inline ddl/column.go]: can be obtained from? from what?

**Comment by zimulala** [comment]
PTAL @winkyao

**Comment by zimulala** [comment]
/run-common-test tidb-test=release-1.0 tidb-private-test=release-1.0 tikv=release-1.0

**Comment by zimulala** [comment]
/run-common-test tidb-test=pr/495 tidb-private-test=release-1.0 tikv=release-1.0

**Comment by zimulala** [comment]
/run-common-test tidb-test=release-1.0 tidb-private-test=release-1.0 tikv=release-1.0

File: ddl/column.go [hunk 4]
```diff
@@ -450,6 +450,9 @@ func (d *ddl) doModifyColumn(t *meta.Meta, job *model.Job, col *model.ColumnInfo
 		return ver, infoschema.ErrColumnNotExists.GenByArgs(oldName, tblInfo.Name)
 	}
 
+	// We need the latest column's offset and state. This information can be obtained from the store.
```
Comment (winkyao) [inline ddl/column.go]: We can't set the offset in ddl.ModifyColumn, because the offset may change when we execute the ddl.doModifyColumn.

**Comment by zimulala** [comment]
/run-common-test tidb-test=pr/498 tidb-private-test=release-1.0 tikv=release-1.0

**Comment by zimulala** [comment]
/run-common-test tidb-test=pr/498 tidb-private-test=release-1.0 tikv=release-1.0

**Comment by zimulala** [comment]
/run-common-test tidb-test=pr/498 tidb-private-test=release-1.0 tikv=release-1.0

**Comment by zimulala** [comment]
/run-common-test tidb-test=pr/498 tidb-private-test=release-1.0 tikv=release-1.0

**Comment by zimulala** [comment]
/run-common-test tidb-test=pr/498 tidb-private-test=release-1.0 tikv=release-1.0

**Comment by zimulala** [comment]
/run-common-test tidb-test=pr/498 tidb-private-test=release-1.0 tikv=release-1.0

**Comment by zimulala** [comment]
/run-common-test tidb-test=pr/498 tidb-private-test=release-1.0 tikv=release-1.0

**Comment by zimulala** [comment]
/run-common-test tidb-test=pr/498 tidb-private-test=release-1.0 tikv=release-1.0

**Comment by zimulala** [comment]
/run-integration-common-test tidb-test=pr/498 tidb-private-test=release-1.0 tikv=release-1.0

**Comment by zimulala** [comment]
/run-common-test tidb-test=pr/498 tidb-private-test=release-1.0 tikv=release-1.0
/run-integration-common-test tidb-test=pr/498 tidb-private-test=release-1.0 tikv=release-1.0

**Pull request merged**